src/Neo/Network/P2P/Payloads/Conditions/AndCondition.cs

```diff
@@ -34,6 +34,30 @@ public class AndCondition : WitnessCondition
         public override int Size => base.Size + Expressions.GetVarSize();
         public override WitnessConditionType Type => WitnessConditionType.And;
 
+        public bool Equals(AndCondition other)
+        {
+            if (ReferenceEquals(this, other))
+                return true;
+            if (other is null) return false;
+            return Type == other.Type &&
+                Size == other.Size &&
+                Expressions.SequenceEqual(other.Expressions);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null) return false;
+            if (obj is not AndCondition ac)
+                return false;
+            else
+                return Equals(ac);
+        }
+
+        public override int GetHashCode()
+        {
+            return (byte)Type;
```

Comment:
Why does this one just return `Type`, while the others use `HashCode.Combine`.

Reply:
`Type` is unique and I it's too much work to loop through all the expressions